src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -591,6 +622,101 @@ protected void deleteNetworks(Azure azureService, List<com.microsoft.azure.manag
         return executeScriptOnVM(vm, instanceScript);
     }
 
+    private Optional<VirtualMachineExtension> retrieveExistingExtension(VirtualMachine vm) {
+
+        if (vm.osType().equals(OperatingSystemTypes.LINUX)) {
+            return vm.listExtensions()
+                     .values()
+                     .stream()
+                     .filter(extension -> extension.publisherName().equals(SCRIPT_EXTENSION_PUBLISHER_LINUX) &&
+                                          extension.typeName().equals(SCRIPT_EXTENSION_TYPE_LINUX))
+                     .findAny();
+        } else if (vm.osType().equals(OperatingSystemTypes.WINDOWS)) {
+            return vm.listExtensions()
+                     .values()
+                     .stream()
+                     .filter(extension -> extension.publisherName().equals(SCRIPT_EXTENSION_PUBLISHER_WINDOWS) &&
+                                          extension.typeName().equals(SCRIPT_EXTENSION_TYPE_WINDOWS))
+                     .findAny();
+        } else {
+            throw new RuntimeException(unsupportedOperatingSystemError(vm.osType().toString()));
+        }
+    }
+
+    private void updateExistingExtension(VirtualMachine vm, VirtualMachineExtension extension, String script) {
+        log.info("Request Azure provider to execute script: " + script);
+        AbstractFuture<VirtualMachine> vmfuture = vm.update()
+                                                    .updateExtension(extension.name())
+                                                    .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, script)
+                                                    .parent()
+                                                    .applyAsync(null);
+        try {
+            vmfuture.get(TIMEOUT_SCRIPT_EXECUTION, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            log.info("An interruption occurred when updating extension to execute script on a VM.");
+            Thread.currentThread().interrupt();
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            log.info("An error occurred when updating extension to execute script on a VM.");
+            e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)